server/basicAuthStatic.js

```diff
@@ -0,0 +1,42 @@
+const _ = require('lodash/fp')
+const reduce = _.reduce.convert({cap: false})
+const find = _.find.convert({cap: false})
+const basicAuth = require('basic-auth')
+
+// Reads auths format from config:
+// Example config:
+// var localjs = {
+//     staticAuths: {
+//         username: 'defaultUser',
+//         password: 'defaultPass',
+//         controller:     { username: 'user', password: 'pass' },
+//         controller2:    {
+//             '*':            { username: 'user', password: 'pass' },
+//             other:          { username: 'usr2', password: 'pas2' },
+//             openMethod:     true,
+//             lockedMethod:   false
+//         }
+//     }
+// }
+
+let getAuth = (url, auths) => {
+  if (!url) return false
+
+  let parts = url.substr(1).split('/');
+  return reduce((memo, part) => {
+    return find((val, key) => {
+      return (new RegExp('^' + part + '$', 'i')).test(key);
+    }, memo) || memo['*'] || memo;
+  }, auths, parts)
```

Comment:
Let's do `parts.substr(1).split('/')` to make `getAuth` a single line function, like:

```
let getAuth = (url, auths) => url && reduce(...`
```